toolchain/parser/parse_node_kind.def

```diff
@@ -75,10 +75,10 @@ CARBON_PARSE_NODE_KIND(PostfixOperator)
 // Struct literals.
 CARBON_PARSE_NODE_KIND(StructLiteral)
 CARBON_PARSE_NODE_KIND(StructTypeLiteral)
+CARBON_PARSE_NODE_KIND(StructLiteralOrTypeLiteralStart)
```

Comment:
Sorry to be a bit late, but I think `StructOrStructTypeLiteralStart` would be more clear here.